sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -249,6 +280,10 @@ object OptimizeIn extends Rule[LogicalPlan] {
           && !v.isInstanceOf[CreateNamedStruct]
           && !newList.head.isInstanceOf[CreateNamedStruct]) {
           EqualTo(v, newList.head)
+        } else if (isInteger(v) && isContinousIntegers(newList.toSet)) {
```

Comment:
Should the number of elements in the list at least be 3?

Reply:
I've added OPTIMIZER_INSET_RANGE_CHECK_THRESHOLD (default to 20) for this purpose